docs/deployment/feature_gates.md

```diff
@@ -46,61 +45,63 @@ The following tables are a summary of the feature gates that you can set on diff
+| CachedRuntimeClients                         | `true`  | `Beta`       | `1.34` | `1.44` |
+| CachedRuntimeClients                         | `true`  | `GA`         | `1.45` | `1.49` |
+| CachedRuntimeClients                         |         | `Removed`    | `1.50` |        |
+| DenyInvalidExtensionResources                | `false` | `Alpha`      | `1.31` | `1.41` |
+| DenyInvalidExtensionResources                | `true`  | `Beta`       | `1.42` | `1.44` |
+| DenyInvalidExtensionResources                | `true`  | `GA`         | `1.45` | `1.49` |
+| DenyInvalidExtensionResources                |         | `Removed`    | `1.50` |        |
+| RotateSSHKeypairOnMaintenance                | `false` | `Alpha`      | `1.28` | `1.44` |
+| RotateSSHKeypairOnMaintenance                | `true`  | `Beta`       | `1.45` | `1.47` |
+| RotateSSHKeypairOnMaintenance (deprecated)   | `false` | `Beta`       | `1.48` | `1.50` |
+| RotateSSHKeypairOnMaintenance (deprecated)   |         | `Removed`    | `1.51` |        |
+| ShootMaxTokenExpirationOverwrite             | `false` | `Alpha`      | `1.43` | `1.44` |
+| ShootMaxTokenExpirationOverwrite             | `true`  | `Beta`       | `1.45` | `1.47` |
+| ShootMaxTokenExpirationOverwrite             | `true`  | `GA`         | `1.48` | `1.50` |
+| ShootMaxTokenExpirationOverwrite             |         | `Removed`    | `1.51` |        |
+| ShootMaxTokenExpirationValidation            | `false` | `Alpha`      | `1.43` | `1.45` |
+| ShootMaxTokenExpirationValidation            | `true`  | `Beta`       | `1.46` | `1.47` |
+| ShootMaxTokenExpirationValidation            | `true`  | `GA`         | `1.48` | `1.50` |
+| ShootMaxTokenExpirationValidation            |         | `Removed`    | `1.51` |        |
+| WorkerPoolKubernetesVersion                  | `false` | `Alpha`      | `1.35` | `1.45` |
+| WorkerPoolKubernetesVersion                  | `true`  | `Beta`       | `1.46` | `1.49` |
+| WorkerPoolKubernetesVersion                  | `true`  | `GA`         | `1.50` | `1.51` |
+| WorkerPoolKubernetesVersion                  |         | `Removed`    | `1.52` |        |
+| DisableDNSProviderManagement                 | `false` | `Alpha`      | `1.41` | `1.49` |
+| DisableDNSProviderManagement                 | `true`  | `Beta`       | `1.50` | `1.51` |
+| DisableDNSProviderManagement                 | `true`  | `GA`         | `1.52` |        |
+| SecretBindingProviderValidation              | `false` | `Alpha`      | `1.38` | `1.50` |
+| SecretBindingProviderValidation              | `true`  | `Beta`       | `1.51` | `1.52` |
+| SecretBindingProviderValidation              | `true`  | `GA`         | `1.53` | `1.54` |
+| SecretBindingProviderValidation              |         | `Removed`    | `1.55` |        |
+| SeedKubeScheduler                            | `false` | `Alpha`      | `1.15` | `1.54` |
+| SeedKubeScheduler                            | `false` | `Deprecated` | `1.55` |        |
```

Comment:
This is not in line with lines 82-83. We should use only one way to list deprecated feature gates.

Reply:
Yes, I also noticed this. I think that the deprecations of `RotateSSHKeypairOnMaintenance` (https://github.com/gardener/gardener/pull/5985) or `ManagedIstio` and `APIServerSNI` (https://github.com/gardener/gardener/pull/6007) are not the most proper ones because they don't update the stage of the feature gate to be `Deprecated`. The corresponding feature gates stay in `Beta` and only have a doc string that they are deprecated. The `ManagedIstio` and `APIServerSNI` case is also different because when you have something deprecated it means "don't enable it, it will be removed" but we rather support and suggest enabling `ManagedIstio` and `APIServerSNI` despite the fact that they are deprecated.

Long story short, I decided to use the `Deprecated` stage instead of only adding a `(deprecated)` next to the feature gate name. I believe that this should be the proper way of deprecating a feature gate. For reference, example for deprecating a feature gate in K8s https://github.com/kubernetes/kubernetes/pull/102966 - they also used the `Deprecated` stage.